src/peft/config.py

```diff
@@ -213,6 +214,21 @@ def _get_peft_type(
         loaded_attributes = cls.from_json_file(config_file)
         return loaded_attributes["peft_type"]
 
+    @classmethod
+    def check_kwargs(cls, **kwargs):
```

Comment:
Let's move this to `OFTConfig`. Here, just return `kwargs` without any further change.

Reply:
@BenjaminBossan What do you mean exactly? Where should I put the check_kwargs?

Reply:
Sorry, let me be more precise.

Inside of `PeftConfigMixin`, let's define:

```python
    @classmethod
    def check_kwargs(cls, **kwargs):
        """Check kwargs before initializing the config instance.

        Subclasses can override this method to add specific checks.

        """
        return kwargs
```

This way, this doesn't do anything for all methods that are not OFT. The `PeftConfigMixin` should not have to worry about OFT implementation details.

Next, inside of `OFTConfig`, we can override this method with the code here:

```python
    @classmethod
    def check_kwargs(cls, **kwargs):
        ... # code from here
        return super().check_kwargs(**kwargs)  # let's also call super just in case
```

Does that make sense?